gateway-server/src/main/java/org/apache/knox/gateway/websockets/ProxyWebSocketAdapter.java

```diff
@@ -198,12 +232,34 @@ public void onError(Throwable cause) {
 
       @Override
       public void onMessageText(String message, Object session) {
-        final RemoteEndpoint remote = getRemote();
-
         LOG.logMessage("[From Backend <---]" + message);
+        remoteLock.lock();
+        final RemoteEndpoint remote = getRemote();
+        if (remote == null) {
+          LOG.logMessage("Remote endpoint is null");
+          if (messageBuffer == null) {
+             messageBuffer = new ArrayList<String>();
+          }
+          if (messageBuffer.size() >= MAX_BUFFER_MESSAGE_COUNT) {
+            throw new RuntimeIOException("Remote is null and message buffer is full. Cannot buffer anymore ");
+          }
+          LOG.logMessage("Buffering message: " + message);
+          messageBuffer.add(message);
+          remoteLock.unlock();
+          return;
+        }
 
         /* Proxy message to frontend */
         try {
+          if (messageBuffer != null) {
+            LOG.logMessage("Found old buffered messages");
+            for(String obj:messageBuffer) {
```

Comment:
This line seems to break our coding style standards (see the missing space after the 'for' keyword); I'm wondering how is it possible it passed static code analysis tools (PMD, Checkstyle)...

Reply:
Fixed with new commit.